src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SessionCategoryMap.java

```diff
@@ -0,0 +1,1028 @@
+    BEREAVEMENT_SUPPORT_PAYMENT_CB_1(BEREAVEMENT_SUPPORT_PAYMENT,CB,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_CC_1(BEREAVEMENT_SUPPORT_PAYMENT,CC,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_CE_1(BEREAVEMENT_SUPPORT_PAYMENT,CE,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_CP_1(BEREAVEMENT_SUPPORT_PAYMENT,CP,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_DD_1(BEREAVEMENT_SUPPORT_PAYMENT,DD,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_LE_1(BEREAVEMENT_SUPPORT_PAYMENT,LE,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_OS_1(BEREAVEMENT_SUPPORT_PAYMENT,OS,false,false,CATEGORY_01,null),
+    BEREAVEMENT_SUPPORT_PAYMENT_RA_1(BEREAVEMENT_SUPPORT_PAYMENT,RA,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_AA_1(WORKING_FAMILIES_TAX_CREDIT,AA,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_CC_1(WORKING_FAMILIES_TAX_CREDIT,CC,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_CE_1(WORKING_FAMILIES_TAX_CREDIT,CE,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_CP_1(WORKING_FAMILIES_TAX_CREDIT,CP,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_CR_1(WORKING_FAMILIES_TAX_CREDIT,CR,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_DD_1(WORKING_FAMILIES_TAX_CREDIT,DD,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_EC_1(WORKING_FAMILIES_TAX_CREDIT,EC,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_EX_1(WORKING_FAMILIES_TAX_CREDIT,EX,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_GC_1(WORKING_FAMILIES_TAX_CREDIT,GC,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_LE_1(WORKING_FAMILIES_TAX_CREDIT,LE,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_OC_1(WORKING_FAMILIES_TAX_CREDIT,OC,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_ON_1(WORKING_FAMILIES_TAX_CREDIT,ON,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_OS_1(WORKING_FAMILIES_TAX_CREDIT,OS,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_OX_1(WORKING_FAMILIES_TAX_CREDIT,OX,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_RA_1(WORKING_FAMILIES_TAX_CREDIT,RA,false,false,CATEGORY_01,null),
+    WORKING_FAMILIES_TAX_CREDIT_RC_1(WORKING_FAMILIES_TAX_CREDIT,RC,false,false,CATEGORY_01,null);
+
+    private final BenefitCode benefitCode;
+    private final Issue issue;
+    private final boolean secondDoctor;
+    private final boolean fqpmRequired;
+    private final SessionCategory category;
+    private final Integer ticketOverride;
+
+    public static final String SERVICE_CODE = "BBA3";
+
+    private static final Map<Integer, SessionCategoryMap> BY_QUERY = new HashMap<>();
+
+    static {
+        for (SessionCategoryMap e: values()) {
+            BY_QUERY.put(getQueryHash(e.benefitCode,e.issue,e.secondDoctor,e.fqpmRequired), e);

```

Comment:
again why don't we pass in the full SessionCategoryMap object to create the hash?